### PR TITLE
Refine arm pl011 upstream

### PR DESCRIPTION
**Pull request by SgrrZhf**
In this PR, I add all available uart_pl011 devices on fvp_baser_aemv8r platforms.

To make use of all available uart_pl011 devices, I refine creating the device instance code of uart_pl011. 
- The old way creates uart_pl011 devices one by one only supports 2 uart_pl011 devices or fewer.
- The new way creates uart_pl011 devices via macro 'DT_INST_FOREACH_STATUS_OKAY' and without the number limit of  uart_pl011.

At last, I remove 'CONFIG_UART_PL011_PORT0=y' from defconfig files because we can select the uart_pl011 device by setting 'status=ok' in the device tree file.

I can't remove 'UART_PL011_PORT0' from Kconfig.pl011 file because this option is used by v2m_musca_s1 board and I don't have this board. Hope someone can remove them fully.
https://github.com/zephyrproject-rtos/zephyr/blob/1d189487b4d5c6cba0be8c2856e738ec44e6562d/boards/arm/v2m_musca_s1/pinmux.c#L44

### CODE REVIEW

**Review by povergoing — DISMISSED** [review]
LGTM, so the `CONFIG_UART_PL011_PORTn` will be defined automatically according to the uart nums and status in dts?

**Comment by SgrrZhf** [comment]
> LGTM, so the `CONFIG_UART_PL011_PORTn` will be defined automatically according to the uart nums and status in dts?

No, Only two boards below select 'CONFIG_UART_PL011_PORTn' in the Kconfig.defconfig file. 
- v2m_musca_b1
https://github.com/zephyrproject-rtos/zephyr/blob/d92e3b88bacc6e6fc1b6222421ef4accdf2d574e/boards/arm/v2m_musca_b1/Kconfig.defconfig#L25
 
- v2m_musca_s1
https://github.com/zephyrproject-rtos/zephyr/blob/d92e3b88bacc6e6fc1b6222421ef4accdf2d574e/boards/arm/v2m_musca_s1/Kconfig.defconfig#L25

Four boards below don't need 'CONFIG_UART_PL011_PORTn' anymore.

- fvp_baser_aemv8r_aarch32
- fvp_base_revc_2xaemv8a
- fvp_baser_aemv8r
- qemu_cortex_a53

File: dts/arm64/fvp-aemv8r/fvp-aemv8r.dtsi [hunk 1]
```diff
@@ -79,5 +79,35 @@
 			label = "UART_0";
 			clocks = <&uartclk>;
 		};
+
+		uart1: uart@9c0a0000 {
```
Comment (carlocaione) [inline dts/arm64/fvp-aemv8r/fvp-aemv8r.dtsi:83]: Is this valid also for `fvp_base_revc_2xaemv8a`? Can you extend that as well?

File: drivers/serial/uart_pl011.c [hunk 2]
```diff
@@ -459,132 +459,65 @@ void pl011_isr(const struct device *dev)
...
-		    &pl011_driver_api);
+#define PL011_IRQ_CONFIG_FUNC_BODY(n, prop, i) \
+	{ \
+		IRQ_CONNECT(DT_IRQ_BY_IDX(n, i, irq), \
+			DT_IRQ_BY_IDX(n, i, priority), \
+			pl011_isr, \
+			DEVICE_DT_GET(n), \
+			0); \
+		irq_enable(DT_IRQ_BY_IDX(n, i, irq)); \
+	}
```
Comment (carlocaione) [inline drivers/serial/uart_pl011.c:471]: nit: please use a common format to align the `\` at the end of the line.

File: dts/arm64/fvp-aemv8r/fvp-aemv8r.dtsi (hunk 1 above)
Comment (SgrrZhf) [reply dts/arm64/fvp-aemv8r/fvp-aemv8r.dtsi:83]: Yes, of course. I can do that.

File: drivers/serial/uart_pl011.c (hunk 2 above)
Comment (SgrrZhf) [reply drivers/serial/uart_pl011.c]: Done

**Review by povergoing — APPROVED** [review]
refresh +1

**Review by carlocaione — APPROVED** [review]

**Review by npitre — APPROVED** [review]

**Pull request merged**